### PR TITLE
ARROW-18264: [Python] Add missing value accessor to temporal types

### PR DESCRIPTION
**Pull request by 0x26res**

### CODE REVIEW

File: python/pyarrow/tests/test_scalars.py [hunk 1]
```diff
@@ -244,14 +244,32 @@ def test_time_from_datetime_time():
...
+    (2**62, pa.timestamp("ns")),
+    (None, pa.timestamp("ns")),
+    (1, pa.duration("ns")),
+    (2**62, pa.duration("ns")),
+    (None, pa.duration("ns")),
+    ((1, 2, -3), pa.month_day_nano_interval()),
+    (None, pa.month_day_nano_interval()),
 ])
-def test_time_values(value, time_type):
+def test_logical_types_values(value, time_type: pa.DataType):
```
Comment (pitrou) [inline python/pyarrow/tests/test_scalars.py:272]: Logical types do not really exist in Arrow (in the sense that all types exist in the same category).
```suggestion
def test_temporal_values(value, time_type: pa.DataType):
```

**Comment by pitrou** [comment]
@jorisvandenbossche Do you want to comment on this?

**Review by jorisvandenbossche — APPROVED** [review]
Looks good, thanks!

**Comment by 0x26res** [comment]
@jorisvandenbossche @pitrou thanks for reviewing 👍

**Pull request merged**